packages/app/src/providers/h5grove/h5grove-api.ts

```diff
@@ -46,12 +52,16 @@ export class H5GroveApi extends ProviderApi {
     assertDataset(entity);
 
     const DTypedArray = typedArrayFromDType(entity.type);
-    if (!DTypedArray) {
-      return this.fetchData(params);
+    if (DTypedArray) {
+      const buffer = await this.fetchBinaryData(params);
+      const array = new DTypedArray(buffer);
+      return hasScalarShape(entity) ? array[0] : [...array];
     }
 
-    const buffer = await this.fetchBinaryData(params);
-    return new DTypedArray(buffer);
+    const value = await this.fetchData(params);
+    return hasArrayShape(entity)
+      ? (value as unknown[]).flat(entity.shape.length - 1)
```

Comment:
There is a complexity layer that you did not take into account there: the presence of `selection` in the `params` changes the shape of the array returned from the request.

This means the checks of shapes must be done on the resulting slice rather than on the full dataset.